pkg/supervisor/supervisor.go

```diff
@@ -88,29 +84,97 @@ func (su *Supervisor) save(data []byte, fd int) {
 	if fd > 0 {
 		su.fd = fd
 	}
-	su.dataStorage.Write(data)
+	su.dataStorage.Save(data)
 }
 
 // Load resources kept by this supervisor
 //  1. daemon runtime states
 //  2. file descriptor
 //
 // Note: the resources should be not be consumed.
-func (su *Supervisor) load(data []byte, oob []byte) (nData uint, nOob int, err error) {
+func (su *Supervisor) load() ([]byte, int, error) {
 	su.mu.Lock()
 	defer su.mu.Unlock()
 
-	if su.fd > 0 {
-		b := syscall.UnixRights(su.fd)
-		nOob = copy(oob, b)
+	data, err := su.dataStorage.Load()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	return data, su.fd, nil
```

Comment:
Should we check if fd is positive?

Reply:
Should be ok, but maybe we don't have to limit it.